src/java/org/apache/cassandra/io/sstable/metadata/MetadataSerializer.java

```diff
@@ -330,27 +349,4 @@ public void updateSSTableMetadata(Descriptor descriptor, Map<MetadataType, Metad
         currentComponents.putAll(updatedComponents);
         rewriteSSTableMetadata(descriptor, currentComponents);
     }
-
-    /**
-     * Read the compression info file pointed by the given descriptor and create the corresponding encryptor.
-     *
-     * Returns null if no encryption applies (version doesn't support it, compression is not applied, or the applicable
-     * compression does not include encryption).
-     */
-    private ICompressor getEncryptor(Descriptor desc)
-    {
-        if (!desc.version.metadataAreEncrypted())
-            return null;
-        File compressionFile = desc.fileFor(Component.COMPRESSION_INFO);
-        if (!compressionFile.exists())
-            return null;
-
-        try (CompressionMetadata cm = CompressionMetadata.read(compressionFile, true))
-        {
-            // Note: we use only the encryption component, without any compression. The reason for doing this is to
-            // avoid having to allocate (and save the size of) an additional buffer to hold the larger uncompressed
-            // serialization on reads.
-            return cm.parameters.getSstableCompressor().encryptionOnly();
-        }
-    }
```

Comment:
are these two commits going to get squashed before merge ?   
(curious as to avoiding changes like this^ appearing in the mainlines)

Reply:
Good point, fixed the previous commit.